### PR TITLE
Upgrade actions in CI

### PR DESCRIPTION
**Pull request by wsor4035**
Add compact, short information about your PR for easier understanding:

- upgrades all the actions use in workflows to resolve warnings

## To do

This PR is a Work in Progress

- [x] verify no warnings when actions finish

## Unresolved issues/upstream needs to handle first

leafo/gh-actions-lua@v10 - https://github.com/leafo/gh-actions-lua/issues/42
leafo/gh-actions-luarocks@v4.3.0 - https://github.com/leafo/gh-actions-luarocks/issues/22

## Outside the scope of just upgrading actions

lukka/run-vcpkg@v7 - https://github.com/minetest/minetest/pull/14377#discussion_r1493006875

## How to test

use your eyes to check actions reports

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -95,15 +95,15 @@ jobs:
...
+      - uses: actions/checkout@v4
 
       - name: Checkout IrrlichtMt
         run: |
           $ref = @(Get-Content misc\irrlichtmt_tag.txt)
           git clone https://github.com/minetest/irrlicht lib\irrlichtmt --depth 1 -b $ref[0]
 
       - name: Restore from cache and run vcpkg
-        uses: lukka/run-vcpkg@v7
+        uses: lukka/run-vcpkg@v11.1
```
Comment (sfan5) [inline .github/workflows/windows.yml:106]: upgrading this one should be possible but it looks like you'll have to move the arguments into a `vcpkg.json`
(please don't put it into the root folder tho)

Comment (wsor4035) [reply .github/workflows/windows.yml]: updated the pr up top to denote out of scope

**Review by sfan5 — APPROVED** [review]

**Comment by sfan5** [comment]
btw the [irrlicht repo](https://github.com/minetest/irrlicht) could use the same treatment.

**Comment by wsor4035** [comment]
> btw the [irrlicht repo](https://github.com/minetest/irrlicht) could use the same treatment.

I'll look this afternoon

**Review by appgurueu — APPROVED** [review]
hmmyes the numbers are bigger now and CI still work so it's probably fine (btw i think this qualifies as trivial so doesn't even need two approvals)

**Pull request merged**